src/app/clusters/network-commissioning/network-commissioning-events.cpp

```diff
@@ -0,0 +1,156 @@
+#include "network-commissioning-events.h"
+#include "network-commissioning.h"
+
+#include <app/util/af.h>
+#include <gen/cluster-id.h>
+#include <gen/command-id.h>
+#include <gen/enums.h>
+
+#include <app/InteractionModelEngine.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <lib/support/Span.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/ConnectivityManager.h>
+
+#include <platform/CHIPDeviceEvent.h>
+
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+#include <platform/ThreadStackManager.h>
+#endif // CHIP_DEVICE_CONFIG_ENABLE_THREAD
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::clusters;
+using namespace chip::app::clusters::NetworkCommissioning;
+using namespace chip::app::clusters::NetworkCommissioning::Internal;
+
+namespace chip {
+namespace app {
+namespace clusters {
+namespace NetworkCommissioning {
+namespace Internal {
+
+chip::NodeId networkCommissionerNodeId = chip::kAnyNodeId;
+ClusterState sClusterState             = ClusterState::kIdle;
+uint8_t sCurrentEnablingNetworkSeq     = kInvalidNetworkSeq;
+SecureSessionHandle sCommissionerSecureSessionHandle;
+
+void SetEnablingNetworkSeq(uint8_t seq)
+{
+    sCurrentEnablingNetworkSeq = seq;
+}
+
+ClusterState GetClusterState()
+{
+    return sClusterState;
+}
+
+void MoveClusterState(ClusterState newState, CHIP_ERROR err)
+{
+    chip::app::CommandSender * commandSender = nullptr;
+    CHIP_ERROR processError                  = CHIP_NO_ERROR;
+    switch (sClusterState)
+    {
+    case ClusterState::kIdle: {
+        switch (newState)
+        {
+        case ClusterState::kEnablingNetwork:
+            // AddEventHandler will do nothing is it is already added.
+            chip::DeviceLayer::PlatformMgr().AddEventHandler(OnNetworkEnableStatusChangeCallback, 0);
+            break;
+        default:
```

Comment:
Is it legal to transition directly from `kIdle` to `kScanningNetwork`?

Reply:
Yes, `kScanningNetwork` is reserved for ScanNetwork command, which will be added later.
The spec does not define detailed behavior of network commissioning cluster, thus this state machine is only used internally.